paimon-core/src/main/java/org/apache/paimon/io/DataFilePathFactory.java

```diff
@@ -43,14 +45,18 @@ public class DataFilePathFactory {
     private final String changelogFilePrefix;
     private final boolean fileSuffixIncludeCompression;
     private final String fileCompression;
+    private final ExternalPathProvider externalPathProvider;
```

Comment:
You should create a new class to this class, for example, DataFileExternalProvider here, which contains `relativePath` already.